AllJoyn/Platform/Bridge/BridgeNET/BridgeNET.cpp

```diff
@@ -67,22 +67,71 @@ namespace BridgeNet
         std::shared_ptr<IBridge> bridge = IBridge::Get();
         bridge->AddAdapter(scriptAdapter);
         bridge->Initialize();
-    
+
         return S_OK;
     }
 
-    void BridgeJs::AddDevice(DeviceInfo^ info, 
-        String^ baseTypeXml, 
-        String^ script, 
+    // Wraps a shared_ptr to an IAdapterIORequest; preserves shared_ptr lifetime management
+    // of the request object and allows waiting for the request to complete.
+    ref class AdapterRequestAwaiter
+    {
+    public:
+        AdapterRequestAwaiter(const std::shared_ptr<IAdapterAsyncRequest>& request)
```

Comment:
Oh, the reason for the extra pointer indirection is because a ref class can't directly contain a non-ref object like a shared_ptr. But it can hold a raw pointer to it, though of course then the std_ptr needs to be deleted explicitly.

I don't see a race condition there, but I'll switch to pass-by-value anyway since there's no need for that micro-optimization.